CONTRIBUTING.md

```diff
@@ -32,20 +32,67 @@ Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).
 - **Channel**: Code is built and tested using the **stable** channel of Rust, but documented and formatted with **nightly** <sup>[*](https://github.com/kube-rs/kube-rs/issues/707)</sup>
 - **Formatting**: To format the codebase, run `make fmt`
 - **Documentation** To check documentation, run `make doc`
-- **Testing**: To run tests, run `make test`.
+- **Testing**: To run tests, run `make test` and see below.
 
 ## Testing
 
-Most tests can be run with `cargo test --all`, but because of features, some tests must be run a little more precisely.
-For the complete variations see the `make test` target in the `Makefile`.
+We have 3 classes of tests.
 
-Some tests and examples require an accessible kubernetes cluster via a `KUBECONFIG` environment variable.
+- Unit tests
+- Integration tests (requires Kubernetes)
+- End to End tests (requires Kubernetes)
 
-- unit tests marked as `#[ignore]` run via `cargo test --all --lib -- --ignored`
-- examples run with `cargo run --example=...`
-- [integration tests](https://github.com/kube-rs/kube-rs/tree/master/integration)
+The last two will try to access the Kubernetes cluster that is your `current-context`; i.e. via your local `KUBECONFIG` evar or `~/.kube/config` file.
+
+The easiest way set up a minimal Kubernetes cluster for these is with [`k3d`](https://k3d.io/).
+
+### Unit Tests
+
+**Most** unit tests are run with `cargo test --lib --doc --all`
+
+but because of feature-sets, doc tests, and examples, you will need a couple of extra invocations to replicate our CI.
```

Comment:
```suggestion
**Most** unit tests are run with `cargo test --lib --doc --all`, but because of feature-sets, doc tests, and examples, you will need a couple of extra invocations to replicate our CI.
```